include/envoy/http/header_map.h

```diff
@@ -263,6 +263,20 @@ class HeaderEntry {
   void value(const char*); // Do not allow auto conversion to std::string
 };
 
+struct HeaderListView {
+  virtual ~HeaderListView() = default;
+
+  /**
+   * @return all of header keys.
+   */
+  virtual std::vector<const HeaderString*> keys() const PURE;
```

Comment:
this looks like the right direction, but if the keys are guaranteed non-null, consider using std::reference_wrapper<const T>(t) here rather than a pointer.

If they may be null then doc that.

Reply:
done